src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -236,7 +236,10 @@ class ConnectedSubchannelStateWatcher
     GRPC_SUBCHANNEL_WEAK_UNREF(subchannel_, "state_watcher");
   }
 
-  void Orphan() override { health_check_client_.reset(); }
+  void Orphan() override {
+    // Must be called while holding subchannel_->mu.
```

Comment:
Nit: Please move this comment to the line above the method, since it's about calling the method, not the implementation of the method.